internal/controller/base_controller.go

```diff
@@ -136,6 +153,22 @@ func (self *BaseController[T]) handleErr(op BaseControllerOp, obj T, err error)
 	return CtrlResultForErr(err)
 }
 
+// ReconcileStatus is a helper function to reconcile the status of an object and requeue on update errors
+func (self *BaseController[T]) ReconcileStatus(ctx context.Context, obj T) (ctrl.Result, error) {
+	log := ctrl.LoggerFrom(ctx)
+
+	patch := client.MergeFrom(obj)
+	if err := self.Kube.Status().Patch(ctx, obj, patch); err != nil {
```

Comment:
I think this will always end up creating an empty patch. The patch needs to be created from the object status in its current state and them modified before calling `.Status().Patch`.

https://sdk.operatorframework.io/docs/building-operators/golang/references/client/#patch